main.go

```diff
@@ -57,12 +58,15 @@ func main() {
 		return
 	}
 
+	var stateReader io.Reader
 	if *stateJSON == "" {
-		panic(errors.New("Mising required '-state' value"))
+		stateReader = bufio.NewReader(os.Stdin)
+	} else {
+		stateReader = strings.NewReader(*stateJSON)
```

Comment:
What do you think about about the following, instead (am I being silly in my aversion `else`?)...

```golang
stateReader := bufio.NewReader(os.Stdin)
if *stateJSON != "" {
  stateReader = strings.NewReader(*stateJSON)
}

Reply:
seems reasonable enough to me... i like it. left-justifies the happy path. i'll make it happen.

Reply:
pushed up a change very similar to the one you suggested, but with explicit interface typing as the two concrete types differ.